app/views/events.py

```diff
@@ -123,6 +123,41 @@ def read_grow(message) -> None:
             {"succeeded": True, "grow": grow.to_json()},
         )
             
+    @socketio.on("search_recipes")
+    def search_recipes(message) -> None:
+        print("search_recipes message:", message)
+        if 'search_name' not in message:
+            send_message_to_namespace_if_specified(
+                socketio,
+                message,
+                "search_recipes_response",
+                {"succeeded": False, "reason": "Search name not included"},
+            )
+            return
+        
+        search_name: str = message['search_name']
+        matching_recipes: List[Recipe] = app_config.db.read_recipes_with_name(search_name)
+
+        recipe_ids: List[int] = []
+        for r in matching_recipes:
+            if not r.recipe_id:
+                print("FATAL ERROR. Null recipe_id:", r.recipe_id)
+            else:
+                recipe_ids.append(r.recipe_id)
+
+        recipe_phases: List[RecipePhase] = app_config.db.read_phases_from_recipes(recipe_ids)
+        
+        recipes_json = [r.to_json() for r in matching_recipes]
+        recipe_phases_json = [rp.to_json() for rp in recipe_phases]
+
+        send_message_to_namespace_if_specified(
+            socketio,
+            message,
+            "search_recipes_response",
+            {"succeeded": True, "recipes": recipes_json, "recipe_phases": recipe_phases_json},
+        )
+        print("search_recipes message here", {"succeeded": True, "recipes": recipes_json, "recipe_phases": recipe_phases_json})
```

Comment:
Nit: Delete this print, don't need a print statement here if we return success to client